components/signup.jsx

```diff
@@ -1,29 +1,11 @@
 import React from 'react';
-import {IntlMixin, FormattedHTMLMessage} from 'react-intl';
+import IntlMixin from 'react-intl';
+import Email from '../components/email-input.jsx';
+import PrivacyPolicy from '../components/privacy-policy-input.jsx';
+import SubmitButton from '../components/submit-button.jsx';
 
 var Form = React.createClass({
-  mixins: [IntlMixin],
-  componentDidMount: function() {
-    var
-        $opt_in = $('label[for="opt_in"]'),
-        $email = $('#email'),
-        donator_email = donator_email || undefined;
-
-    $opt_in.find('label').prepend('<span className="required">* ');
-
-    $email
-        .attr('aria-label', 'Email Address');
-
-    if (donator_email !== undefined) {
-      $email.val(donator_email).attr('value',donator_email);
-    }
-
-    $('#signup').parsley().subscribe("parsley:form:validate", function (formInstance) {
-      if (!formInstance.isValid()) {
-        formInstance.submitEvent.preventDefault();
-      }
-    });
-  },
+  mixins: [IntlMixin, require('../mixins/form.jsx')],
```

Comment:
This whole file is updated to use the email and privacy components, along with a submit button component.